x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheService.java

```diff
@@ -251,51 +277,62 @@ boolean isCacheFileToSync(CacheFile cacheFile) {
      * non empty set of completed ranges this method also fsync the shard's snapshot cache directory, which is the parent directory of the
      * cache entry. Note that cache files might be evicted during the synchronization.
      */
-    protected synchronized void synchronizeCache() {
-        long count = 0L;
-        final Set<Path> cacheDirs = new HashSet<>();
-        final long startTimeNanos = threadPool.relativeTimeInNanos();
-        final long maxCacheFilesToSync = Math.min(numberOfCacheFilesToSync.get(), this.maxCacheFilesToSyncAtOnce);
-        for (long i = 0L; i < maxCacheFilesToSync; i++) {
+    protected void synchronizeCache() {
+        cacheSyncLock.lock();
+        try {
             if (lifecycleState() != Lifecycle.State.STARTED) {
```

Comment:
I would prefer to keep this inside the loop to ensure we break out as soon as possible when shutting down.

Reply:
Ok